lib/features/family/features/home_screen/presentation/pages/family_home_screen.dart

```diff
@@ -0,0 +1,199 @@
+import 'dart:async';
+
+import 'package:flutter/material.dart';
+import 'package:flutter_bloc/flutter_bloc.dart';
+import 'package:flutter_svg/flutter_svg.dart';
+import 'package:givt_app/core/enums/amplitude_events.dart';
+import 'package:givt_app/features/children/shared/profile_type.dart';
+import 'package:givt_app/features/family/app/family_pages.dart';
+import 'package:givt_app/features/family/app/injection.dart';
+import 'package:givt_app/features/family/features/home_screen/cubit/family_home_screen_cubit.dart';
+import 'package:givt_app/features/family/features/home_screen/presentation/models/family_home_screen.uimodel.dart';
+import 'package:givt_app/features/family/features/home_screen/presentation/pages/family_home_overlay.dart';
+import 'package:givt_app/features/family/features/home_screen/widgets/give_button.dart';
+import 'package:givt_app/features/family/features/home_screen/widgets/gratitude_game_button.dart';
+import 'package:givt_app/features/family/features/profiles/cubit/profiles_cubit.dart';
+import 'package:givt_app/features/family/features/reflect/presentation/models/grateful_avatar_bar_uimodel.dart';
+import 'package:givt_app/features/family/features/reflect/presentation/widgets/grateful_avatar_bar.dart';
+import 'package:givt_app/features/family/shared/design/components/components.dart';
+import 'package:givt_app/features/family/shared/widgets/loading/custom_progress_indicator.dart';
+import 'package:givt_app/features/family/shared/widgets/texts/shared_texts.dart';
+import 'package:givt_app/features/family/utils/family_auth_utils.dart';
+import 'package:givt_app/shared/widgets/base/base_state_consumer.dart';
+import 'package:givt_app/shared/widgets/fun_scaffold.dart';
+import 'package:givt_app/utils/analytics_helper.dart';
+import 'package:go_router/go_router.dart';
+
+class FamilyHomeScreen extends StatefulWidget {
+  const FamilyHomeScreen({super.key});
+
+  @override
+  State<FamilyHomeScreen> createState() => _FamilyHomeScreenState();
+}
+
+class _FamilyHomeScreenState extends State<FamilyHomeScreen> {
+  OverlayEntry? overlayEntry;
+  bool overlayVisible = false;
+  final _cubit = getIt<FamilyHomeScreenCubit>();
+
+  @override
+  void didChangeDependencies() {
+    super.didChangeDependencies();
+
+    context.read<ProfilesCubit>().fetchAllProfiles();
+    _cubit.init();
+  }
+
+  @override
+  void dispose() {
+    if (overlayEntry != null && overlayEntry!.mounted) {
+      closeOverlay();
+    }
+
+    super.dispose();
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    return BaseStateConsumer(
+      cubit: _cubit,
+      onLoading: (context) => const CustomCircularProgressIndicator(),
```

Comment:
I was thinking, we should probably just make this the default of the onLoading rather than the CircularProgressIndicator.